chain/chain/src/resharding/resharding_actor.rs

```diff
@@ -1,23 +1,77 @@
-use super::types::FlatStorageSplitShardRequest;
-use near_async::messaging::{self, Handler};
+use super::types::ReshardingRequest;
+use crate::flat_storage_resharder::{FlatStorageResharder, FlatStorageReshardingTaskStatus};
+use crate::ChainStore;
+use near_async::futures::{DelayedActionRunner, DelayedActionRunnerExt};
+use near_async::messaging::{self, HandlerWithContext};
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::BlockHeight;
+use near_store::{ShardUId, Store};
+use time::Duration;
 
 /// Dedicated actor for resharding V3.
-pub struct ReshardingActor {}
+pub struct ReshardingActor {
+    chain_store: ChainStore,
+}
 
 impl messaging::Actor for ReshardingActor {}
 
-impl Handler<FlatStorageSplitShardRequest> for ReshardingActor {
-    fn handle(&mut self, msg: FlatStorageSplitShardRequest) {
-        self.handle_flat_storage_split_shard_request(msg);
+impl HandlerWithContext<ReshardingRequest> for ReshardingActor {
+    fn handle(&mut self, msg: ReshardingRequest, ctx: &mut dyn DelayedActionRunner<Self>) {
+        match msg {
+            ReshardingRequest::FlatStorageSplitShard { resharder } => {
+                self.handle_flat_storage_split_shard(resharder);
+            }
+            ReshardingRequest::FlatStorageShardCatchup {
```

Comment:
These are three different types of requests. We should not group them under `ReshardingRequest` but instead separate these into three different Messages.

Reply:
I took inspiration from other requests that are using `enum` to group requests. 
I can make three different requests and group the resulting three senders into a struct. Can you explain what would be the advantage compared to this approach?

Reply:
Yeah, no, it's just additional boilerplate code at the call site and at the end of the day all requests are effectively different. We've already defined them to be part of resharding as we are messaging to the resharding actor.

We've made this mistake a long time ago in ShardsManager already (NetworkRequest is still fine but I would ideally like to separate NetworkRequest from PeerManagerAdapter too). Anyway, I don't think we follow this enum pattern for new code.

Reply:
Fair enough, I'll change the enum into separate messages

Reply:
Done!